orig_index/web.py

```diff
@@ -46,25 +53,53 @@ def archive_hash(hash: str):
     return api_explore_files_in_archive(hash)
 
 
-@APP.get("/api/normalized/hash/{hash}")
-def normalized_detail(hash: str):
+@APP.get("/api/normalized/hash/{hash}", response_class=HTMLResponse)
+def normalized_detail(hash: str, request: Request):
     """
     Serves the text of the snippets, and mentions what the "oldest" archive
     containing this normalized file is.
 
     Calculating hash-matches of snippets or embedding-similarity of matches is a
     lot more expensive, and should lazy-load from other endpoints.
     """
-    return api_normalized_detail(hash)
+    results = api_normalized_detail(hash)
+    return templates.TemplateResponse(
+        "index.html", {"request": request, "results": results}, block_name="results"
+    )
 
 
-@APP.get("/api/normalized/partial/{hash}")
-def normalized_partial(hash: str):
+@APP.get("/api/normalized/partial/{hash}", response_class=HTMLResponse)
+def normalized_partial(hash: str, request: Request):
     """
     Search for hashes of snippets of this normalized file, assuming that the
     snippets are unmodified.
     """
-    return api_normalized_partial(hash)
+    results = api_normalized_detail(hash)
+    partial = api_normalized_partial(hash)
+    snippet_table = "<table border='1'>\n"
+    snippet_table += "<tr><th>Source</th>"
+
+    for col in partial["found"]:
+        snippet_table += f"<th><a href='{request.url_for('normalized_partial', hash=col['hash'])}' title='{col['hash']}'>{col['hash'][:4]}...</a></th>"
+
+    snippet_table += "</tr>\n"
+    for i, snip in enumerate(results["snippets"]):
+        snippet_table += "<tr>"
+        snippet_table += (
+            "<td><pre style='white-space: pre-wrap'>"
+            + html.escape(snip["text"])
+            + "</pre></td>"
+        )
+        for col in partial["found"]:
+            snippet_table += "<td>" + ("X" if i in col["incl"] else "") + "</td>"
+        snippet_table += "</tr>\n"
+    snippet_table += "</table>"
+
+    return templates.TemplateResponse(
+        "index.html",
+        {"request": request, "results": results, "snippet_table": snippet_table},
+        block_name="results",
+    )
```

Comment:
It's generally not a good idea to string together the html fragments here. We can talk about how to move this to the template itself under a different block name.

Reply:
Yeah, so right now for normalized files we have "hash matches" (don't need sidebar), "partial matches" (need sidebar, binary?), and "close matches" using embedding distance, which I haven't figured out the display for (but probably more analog in a sidebar).  This code is hacky but should only be here for a day or two.